### PR TITLE
feat: extract rpc<>revm specifc code

### LINKED ISSUE #1689: Extract rpc<->revm specific code from rpc
### Describe the feature

atm some code in https://github.com/paradigmxyz/reth/blob/main/crates/rpc/rpc/src/eth/api/call.rs

like:

https://github.com/paradigmxyz/reth/blob/d164a16be7481ebeeda70bad0069a84bb11163c3/crates/rpc/rpc/src/eth/api/call.rs#L391 

is useful on its own and should not be tied to the rpc impl.

it should be moved either to:
a) rpc-type with feature gated revm dep 
b) revm with featured gated rpc dep

I think, the former makes more sense.

The main motivation for this is, so that other consumers of the rpc-types can get access to the revm helpers as well (e.g. anvil eventually)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by chirag-bgh**
Closes #1689

### CODE REVIEW

File: crates/rpc/rpc-types/src/eth/error.rs [hunk 1]
```diff
@@ -21,3 +27,51 @@ impl EthRpcErrorCode {
         }
     }
 }
+
+#[derive(Debug, thiserror::Error, Clone, PartialEq, Eq)]
+#[allow(missing_docs)]
+pub enum RevmError {
```
Comment (onbjerg) [inline crates/rpc/rpc-types/src/eth/error.rs:33]: This should rly be using the executor error enum instead of writing its own: https://github.com/paradigmxyz/reth/blob/d774d914ceb76ee7d896f348f20e20d1ece5d92f/crates/interfaces/src/db.rs#L3

File: crates/interfaces/src/error.rs [hunk 2]
```diff
@@ -19,4 +19,13 @@ pub enum Error {
...
+
+    #[error(transparent)]
+    Revm(#[from] reth_rpc_types::error::RevmError),
+}
+
+impl From<Error> for reth_rpc_types::error::RevmError {
+    fn from(_: Error) -> Self {
+        Self::Internal
+    }
 }
```
Comment (mattsse) [inline crates/interfaces/src/error.rs:31]: why is this needed, 
bidirectional conversion seems bad

File: crates/rpc/rpc-types/src/eth/error.rs (hunk 1 above)
Comment (mattsse) [reply crates/rpc/rpc-types/src/eth/error.rs:33]: > Ideally the RPC would use an ExecutorFactory like the rest of the code

RPC does not write to Disk and needs to configure the env directly depending on the RPC call.

The EvmEnvProvider trait is used by RPC to get block and cfg of the env and `create_txn_env` configures the txenv based on the request.

**Review by mattsse — COMMENTED** [review]
After some discussion, we want the rpc types to be a completely independent crate so that it can be easily reused throughout the ecosystem and does not come with any additional feature, therefore the issue in question is no longer relevant and needs to be re-evaluated.

Very sorry for wasting your time on this...
